app/models/favorite_handler.rb

```diff
@@ -0,0 +1,13 @@
+class FavoriteHandler < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :post
+
+  validates :user_id, presence: true
+  validates :post_id, presence: true, uniqueness: { scope: :user_id,
+    message: "has already been favorited" }
+
+  def self.destroy_favorite(user_id, post_id)
+    favorite = where('user_id = ? AND post_id = ?', user_id, post_id).first
```

Comment:
`where(...).first` is almost always a bad idea. If you have to do something like that, use `find_by` instead.